randomActivityGen.py

```diff
@@ -229,21 +233,23 @@ def main():
     stats = ET.parse(args["--stat-file"])
     verify_stats(stats)
 
-    # Scale and octave seems like sane values for the moment
-    apply_network_noise(net, stats)
+    centre = find_city_centre(net) if args["--centre.pos"] == "auto" else tuple(map(int, args["--centre.pos"].split(",")))
+
+    # Populate network with street data
+    apply_network_noise(net, stats, centre, float(args["--centre.pop-weight"]), float(args["--centre.work-weight"]))
 
     setup_city_gates(net, stats, int(args["--gates.count"]))
 
     if args["--schools.count"] == "auto":
-        setup_schools(net, stats, None)
+        setup_schools(net, stats, None, centre)
     else:
-        setup_schools(net, stats, int(args["--schools.count"]))
+        setup_schools(net, stats, int(args["--schools.count"]), centre)
```

Comment:
Why is centre being added to school functions?

Reply:
Because the placement of schools relies on the population noise which relies on the location of the city centre.